### PR TITLE
login form  with functioning and google Authentication

### PR DESCRIPTION
**Pull request by Subhajit8085277414**

### CODE REVIEW

File: react-frontend/src/App.js [hunk 1]
```diff
@@ -1,24 +1,12 @@
 import React from "react";
-import logo from "./logo.svg";
 import "./App.css";
+import "./components/MainComponent";
```
Comment (sahilbhatia) [inline react-frontend/src/App.js:3]: We need to use "absolute imports". i.e. start your import path from directory in "src" without a "." or ".." prefix.

e.g. `import "components/MainComponent";`

File: react-frontend/src/components/LoginComponent.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import React, { useState } from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
```
Comment (sahilbhatia) [inline react-frontend/src/components/LoginComponent.js:2]: Please remove it.

1. Individual components should not import bootstrap CSS.
2. We have already imported bootstrap CSS at a common place, hence it's not required.

File: react-frontend/src/components/LoginComponent.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import {
+  Col,
+  Form,
+  Input,
+  Button,
+  Container,
+  FormGroup,
+  Label,
+  InputGroup,
+} from "reactstrap";
```
Comment (sahilbhatia) [inline react-frontend/src/components/LoginComponent.js:13]: All of us need to use common package. We are already using `react-bootstrap`. Check `package.json` file

File: react-frontend/src/components/SignInComponent.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        name={name}
+        email={email}
+        url={url}
+        token={accesstoken}
+      />
+    );
+  }
+  return (
+    <GoogleLogin
+      clientId="270417110073-i9o1p7pf9isq4e9g2mrve1ofrvlaa7hh.apps.googleusercontent.com"
```
Comment (sahilbhatia) [inline react-frontend/src/components/SignInComponent.js:31]: Any sensitive information like "passwords", "client id", etc should go in `.env.local` file.

File: react-frontend/package.json [hunk 5]
```diff
@@ -2,16 +2,17 @@
...
   "dependencies": {
     "@testing-library/jest-dom": "^4.2.4",
-    "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
     "bootstrap": "^4.4.1",
     "react": "^16.13.1",
     "react-bootstrap": "^1.0.0",
     "react-dom": "^16.13.1",
+    "react-google-login": "^5.1.3",
+    "react-router-dom": "^5.1.2",
```
Comment (mayuriardad) [inline react-frontend/package.json:15]: take a rebase with master

**Pull request closed without merging**